File: backend/src/models/ResponseOption.js
const mongoose = require('mongoose');

const responseOptionSchema = new mongoose.Schema(
  {
    label: { type: String, required: true, unique: true },
    description: { type: String, default: '' },
    hours: { type: Number, required: true },
  },
  { timestamps: true }
);

module.exports = mongoose.model('ResponseOption', responseOptionSchema);

File: frontend/lib/models/response_option.dart
class ResponseOption {
  const ResponseOption({
    required this.id,
    required this.label,
    required this.hours,
    this.description,
  });

  factory ResponseOption.fromJson(Map<String, dynamic> json) {
    return ResponseOption(
      id: json['_id'] as String,
      label: json['label'] as String? ?? '',
      hours: (json['hours'] as num?)?.toInt() ?? 0,
      description: json['description'] as String?,
    );
  }

  final String id;
  final String label;
  final int hours;
  final String? description;
}

